libs/reactivex-test/Observable_MaxOperatorTests.cs

```diff
@@ -0,0 +1,250 @@
+using System;
+using System.Collections;
+using System.Threading.Tasks;
+using Cusco.Dispatch;
+using Moq;
+
+namespace Cusco.ReactiveX.Test;
+
+public sealed partial class ObservableTests
+{
+  [Test]
+  public async Task Observable_Max_LengthComparer_ShouldNextThenComplete()
+  {
+    // arrange
+    var observerMock = new Mock<IObserver<string>>().SetupAllProperties();
+    var observer = observerMock.Object;
+
+    // act
+    var observable = Observable.Create<string>(DispatchQueue.main, observer =>
+      {
+        observer.OnNext("abce");
+        observer.OnNext("abc");
+        observer.OnNext("abcd");
+        observer.OnCompleted();
+        return DummyDisposable.instance;
+      })
+      .Max(new LengthStringComparer());
+    observable.Subscribe(observer);
+
+    await observable.LastOrDefaultAsFuture();
+
+    // assert
+    CallSequence.ForMock(observerMock)
+      .VerifyInvocation(observer => observer.OnNext, "abce")
```

Comment:
@abidon Here Max operator return the first max value , maybe we want the last max value? (`abcd` instead of `abce`) 🤔

Reply:
I would say that this is not the responsibility of the operator, but the responsibility of the comparer.